packages/slice-machine/lib/builders/CustomTypeBuilder/Layout/Header/primaryButton.tsx

```diff
@@ -0,0 +1,80 @@
+import { useSelector } from "react-redux";
+import { SliceMachineStoreType } from "@src/redux/type";
+import { Button, Spinner } from "theme-ui";
+import { CustomTypeStatus } from "@src/modules/selectedCustomType/types";
+import {
+  selectCustomTypeStatus,
+  selectIsCurrentCustomTypeHasPendingModifications,
+} from "@src/modules/selectedCustomType";
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+
+interface PrimaryButtonProps {
+  isSavingCustomType: boolean;
+  isPushingCustomType: boolean;
+}
+
+export const PrimaryButton: React.FC<PrimaryButtonProps> = ({
+  isSavingCustomType,
+  isPushingCustomType,
+}) => {
+  const { hasPendingModifications, customTypeStatus } = useSelector(
+    (store: SliceMachineStoreType) => ({
+      hasPendingModifications:
+        selectIsCurrentCustomTypeHasPendingModifications(store),
+      customTypeStatus: selectCustomTypeStatus(store),
+    })
+  );
+  const { saveCustomType, pushCustomType } = useSliceMachineActions();
+
+  const buttonProps = (() => {
```

Comment:
I've not seen that being done before, a function to make the props and children for an element

Reply:
this was already like that, all I did was copy paste it into another file since the original one was getting too big. Can refactor to change it tho

Reply:
updated